### PR TITLE
Fix js callback hell example errors

### PR DESCRIPTION
**Pull request by RyanBard**
* Fixes wrong param passed to putBeefBetweenBuns (cooked beef instead of raw)
* Fixes syntax error in makeBurger callback (mix of function and arrow syntax)
* Fixes typo: leftFright -> leftFridge
* Fixes typo: putBeefinOven -> putBeefInOven
* Changes promise example to nest instead of having getBuns take in beef
* Fixes param order of putBeefBetweenBuns in async/await examples

Signed-off-by: Ryan Bard <john.ryan.bard@gmail.com>

### CODE REVIEW

**Comment by RyanBard** [comment]
Here is some scaffolding to test the first solution (nested callback hell):

```js
function getBeef(cb) {
    setTimeout(() => cb('raw-beef'), 100)
}

function cookBeef(beef, cb) {
    setTimeout(() => {
        if (beef === 'raw-beef') {
            cb('cooked-beef')
            return;
        }
        cb(''); // Using a falsy value since your examples aren't err first callbacks
    }, 100);
}

function getBuns(cb) {
    setTimeout(() => cb('buns'), 100)
}

function putBeefBetweenBuns(buns, beef, cb) {
    setTimeout(() => {
        if (buns !== 'buns') {
            cb('messy-burger')
            return;
        }
        if (beef !== 'cooked-beef') {
            cb('food-poisoning')
            return;
        }
        cb('yummy-burger')
    }, 100)
}

function serve(burger) {
    console.log('after %s ms, serving:', Date.now() - startTime, burger);
}

let startTime = Date.now();
```

Here is some scaffolding to test the third solution (promises):

```js
function getBeef() {
    return new Promise((resolve, reject) => setTimeout(() => resolve('raw-beef'), 100))
}

function cookBeef(beef) {
    return new Promise((resolve, reject) => {
        setTimeout(() => {
            if (beef === 'raw-beef') {
                resolve('cooked-beef')
            } else {
                reject('Invalid: ' + beef)
            }
        }, 100)
    })
}

function getBuns() {
    return new Promise((resolve, reject) => setTimeout(() => resolve('buns'), 100))
}

function putBeefBetweenBuns(buns, beef) {
    return new Promise((resolve, reject) => {
        setTimeout(() => {
            if (buns !== 'buns') {
                reject('messy-burger')
                return
            }
            if (beef !== 'cooked-beef') {
                reject('food-poisoning')
                return
            }
            resolve('yummy-burger')
        }, 100)
    })
}

function serve(burger) {
    console.log('after %s ms, serving:', Date.now() - startTime, burger)
}

let startTime = Date.now()
```

And here is some for the fourth solution (async-await):

```js
async function getBeef() {
    return new Promise((resolve, reject) => setTimeout(() => resolve('raw-beef'), 100))
}

async function cookBeef(beef) {
    return new Promise((resolve, reject) => {
        setTimeout(() => {
            if (beef === 'raw-beef') {
                resolve('cooked-beef')
            } else {
                reject('Invalid: ' + beef)
            }
        }, 100)
    })
}

async function getBuns() {
    return new Promise((resolve, reject) => setTimeout(() => resolve('buns'), 100))
}

async function putBeefBetweenBuns(buns, beef) {
    return new Promise((resolve, reject) => {
        setTimeout(() => {
            if (buns !== 'buns') {
                reject('messy-burger')
                return
            }
            if (beef !== 'cooked-beef') {
                reject('food-poisoning')
                return
            }
            resolve('yummy-burger')
        }, 100)
    })
}

function serve(burger) {
    console.log('after %s ms, serving:', Date.now() - startTime, burger)
}

let startTime = Date.now()
```

File: src/posts/2019-05-08-nested-callbacks.md [hunk 1]
```diff
@@ -222,8 +221,7 @@ If you take advantage of the single-argument style with promises, you can tweak
 const makeBurger = () => {
   return getBeef()
     .then(cookBeef)
-    .then(getBuns)
-    .then(putBeefBetweenBuns)
+    .then(cookedBeef => getBuns().then(buns => putBeefBetweenBuns(buns, cookedBeef)))
```
Comment (zellwk) [inline src/posts/2019-05-08-nested-callbacks.md]: I'd prefer to keep this as the original. IMO, not a good practice to have nested `then` calls. In this case I'd return an array `[cookedBeef, buns]`, so I can use it in the next `then`. But there's no need to explain that at this point. 

Can you remove the changes here please?

File: src/posts/2019-05-08-nested-callbacks.md [hunk 2]
```diff
@@ -207,8 +207,7 @@ Promises can make callback hell much easier to manage. Instead of the nested cod
 const makeBurger = () => {
   return getBeef()
     .then(beef => cookBeef(beef))
-    .then(cookedBeef => getBuns(beef))
-    .then(bunsAndBeef => putBeefBetweenBuns(bunsAndBeef))
+    .then(cookedBeef => getBuns().then(buns => putBeefBetweenBuns(buns, cookedBeef)))
```
Comment (zellwk) [inline src/posts/2019-05-08-nested-callbacks.md]: Same comment as below.

**Comment by RyanBard** [comment]
Updated test functions for the promise solution (since getBuns takes in beef and echos it in the results and putBeefBetweenBuns expects the array):

```js
function getBuns(beef) {
    return new Promise((resolve, reject) => setTimeout(() => resolve(['buns', beef]), 100))
}

function putBeefBetweenBuns([buns, beef]) {
    return new Promise((resolve, reject) => {
        setTimeout(() => {
            if (buns !== 'buns') {
                reject('messy-burger')
                return
            }
            if (beef !== 'cooked-beef') {
                reject('food-poisoning')
                return
            }
            resolve('yummy-burger')
        }, 100)
    })
}
```

**Comment by zellwk** [comment]
Amazing work, @RyanBard. Thanks so much!

**Pull request merged**